lib/avro_turf/cached_confluent_schema_registry.rb

```diff
@@ -1,12 +1,20 @@
 require 'avro_turf/confluent_schema_registry'
+require 'avro_turf/in_memory_cache'
+require 'avro_turf/disk_cache'
 
 # Caches registrations and lookups to the schema registry in memory.
 class AvroTurf::CachedConfluentSchemaRegistry
 
-  def initialize(upstream)
+  # Instantiate a new CachedConfluentSchemaRegistry instance with the given configuration.
+  # By default, uses a provided InMemoryCache to prevent repeated calls to the upstream registry.
+  #
+  # upstream  - The upstream schema registry object that fully responds to all methods in the
+  #             AvroTurf::ConfluentSchemaRegistry interface.
+  # disk_path - Optional path on disk to use the provided DiskCache instead of the default InMemoryCache
+  # cache     - Optional user provided Cache object that responds to all methods in the AvroTurf::InMemoryCache interface.
+  def initialize(upstream, disk_path: nil, cache: nil)
```

Comment:
I don't think this class should know about `disk_path`. I'd rather just have it accept `cache`, e.g.

```ruby
cache = AvroTurf::DiskCache.new(disk_path)
registry = AvroTurf::CachedConfluentSchemaRegistry.new(..., cache: cache)
...
```

Reply:
Sure thing. In retrospect, putting that disk_path stuff in there did seem a bit awkward.